lib/vault/encrypted_model.rb

```diff
@@ -297,11 +344,15 @@ def __vault_persist_attribute!(attribute, options)
         # Generate context if needed
         generated_context = __vault_generate_context(context)
 
-        # Generate the ciphertext and store it back as an attribute
-        ciphertext = Vault::Rails.encrypt(
-          path, key, plaintext,
-          context: generated_context
-        )
+        if transform
+          ciphertext = plaintext
```

Comment:
This bit seems a little counter-intuitive at first glance, can you add a comment as to why it's doing this?

Reply:
IIUC we perform FPE by calling `serializer.encode` above, and we only have a serializer for transform. Could we call it in here instead?
```
if transform
  if selializer
    ciphertext = serializer.encode(plaintext)
  end
else
...
```
This make is a bit more straightforward to read, and also ensures that we never return the plaintext value as the encoded value if the serializer is somehow nil.

Reply:
It looks like a custom serializer can also be passed in so it's not specific to transform (thus we want to call it outside), though the concern of the plaintext assigned this way possibly leaking if the serializer is somehow nil still holds.

Reply:
The serializer can't be nil in these cases because when you include the transform_secrets hash, it creates a serializer as part of the code. If you attempt to give a serializer while also using a transform_secret, it errors out.

Reply:
I see. I was trying to be clever and see if we could avoid the confusing assignment but I guess the added comment would be sufficient for now.